pkg/ipam/trench/watcher.go

```diff
@@ -21,48 +21,28 @@ import (
 	"io"
 	"time"
 
-	"github.com/go-logr/logr"
 	nspAPI "github.com/nordix/meridio/api/nsp/v1"
 	"github.com/nordix/meridio/pkg/ipam/types"
 	"github.com/nordix/meridio/pkg/log"
 	"github.com/nordix/meridio/pkg/retry"
-	"google.golang.org/grpc"
 )
 
-type ConduitWatcher struct {
-	Ctx                        context.Context
-	TrenchWatchers             []TrenchWatcher
-	TrenchName                 string
-	ConfigurationManagerClient nspAPI.ConfigurationManagerClient
-	logger                     logr.Logger
-}
-
+//go:generate mockgen -source=watcher.go -destination=mocks/watcher.go -package=mocks
 type TrenchWatcher interface {
 	AddConduit(ctx context.Context, name string) (types.Conduit, error)
 	RemoveConduit(ctx context.Context, name string) error
 }
 
-func NewConduitWatcher(ctx context.Context, nspConn *grpc.ClientConn, trenchName string, trenchWatchers []TrenchWatcher) (*ConduitWatcher, error) {
-	configurationManagerClient := nspAPI.NewConfigurationManagerClient(nspConn)
+func NewConduitWatcher(ctx context.Context, configurationManagerClient nspAPI.ConfigurationManagerClient, trenchName string, trenchWatchers []TrenchWatcher) {
+	currentConduits := map[string]struct{}{}
 
-	cw := &ConduitWatcher{
-		Ctx:                        ctx,
-		TrenchName:                 trenchName,
-		TrenchWatchers:             trenchWatchers,
-		ConfigurationManagerClient: configurationManagerClient,
-		logger:                     log.FromContextOrGlobal(ctx).WithValues("class", "ConduitWatcher"),
-	}
-	return cw, nil
-}
-
-func (cw *ConduitWatcher) Start() {
-	err := retry.Do(func() error {
+	_ = retry.Do(func() error {
```

Comment:
maybe some logs if the retry loop stops could be helpful (printing err as well)

Reply:
I just added new logs in the retry loop.
The retry loop won't return any error,  `WithErrorIngnored` is used.